libraries/core_libs/network/src/tarcap/packets_handlers/votes_sync_packet_handler.cpp

```diff
@@ -36,11 +37,42 @@ void VotesSyncPacketHandler::process(const PacketData &packet_data, const std::s
     return;
   }
 
+  std::unordered_map<blk_hash_t, uint64_t> voted_value_round;
   std::vector<std::shared_ptr<Vote>> next_votes;
+
   const auto next_votes_count = packet_data.rlp_.itemCount();
+  // Parse and mark all votes as known for sender peer to avoid resends to it
   for (size_t i = 0; i < next_votes_count; i++) {
     auto next_vote = std::make_shared<Vote>(packet_data.rlp_[i].data().toBytes());
-    const auto next_vote_hash = next_vote->getHash();
+    peer->markVoteAsKnown(next_vote->getHash());
+    next_votes.push_back(next_vote);
```

Comment:
I guess pushing all votes here and then removing is not ideal

Reply:
What is better way to do it then? I thought it is the best one :)

Reply:
what was here before